packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -183,21 +223,51 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
 
         plugin.delete(
                 instance,
-                Consumer {
+                {
                     LOG.debug("Deleted item: " + it.item().toString())
                     handler.post { flutterResult.success(null) }
                 },
-                Consumer {
+                {
                     LOG.debug("Deletion Failed: " + it)
                     if (it is DataStoreException && it.localizedMessage == "Wanted to delete one row, but deleted 0 rows.") {
-                        handler.post{ flutterResult.success(null) }
+                        handler.post { flutterResult.success(null) }
                     } else {
-                        prepareError(flutterResult, it, FlutterDataStoreFailureMessage.AMPLIFY_DATASTORE_DELETE_FAILED.toString())
+                        postFlutterError(
+                                flutterResult,
+                                FlutterDataStoreFailureMessage.AMPLIFY_DATASTORE_DELETE_FAILED.toString(),
+                                createErrorMap(it))
                     }
                 }
         )
     }
 
+    @VisibleForTesting
+    fun onSetupObserve(result: Result) {
+        val plugin = Amplify.DataStore.getPlugin("awsDataStorePlugin") as AWSDataStorePlugin
+
+        plugin.observe(
+                       { cancelable ->
+                           LOG.info("Established a new stream form flutter $cancelable")
+                           observeCancelable = cancelable
+                       },
+                       { event ->
+                           LOG.debug("Received event: $event")
+                           if(event.item() is SerializedModel) {
```

Comment:
Huh, interesting. Does this work as expected, while testing?

Reply:
Yes, with the _modelless_ `observe` API, I'm getting events for system models as well. This check then filters it down to SerializedModels only.